### ISSUE TITLE
weight_dict

### ISSUE THREAD

**Issue by Rzx520** [orrzohar/PROB#32]
I noticed while reading the code that you set 'loss_obj_ll(weight_dict)' to 0.0008. How did you set this value?@orrzohar

**Comment by Rzx520** [comment]
If I remember correctly, you said that it seems like you actually gave 300 predictions, but in reality, you only took the first 100 sorted predictions (top_k=100). So why is it that your parameter settings set num_queries to 100 ('-- num_queries', default=100)? I see that previous work has been set to 300, such as OW-DETR and D-DETR.@orrzohar

**Comment by orrzohar** [comment]
Hi @Rzx520,

a) re. loss_obj_ll, you can think of this as a hyperparameter tuning -- but there is actually a mathematical way of getting to this order of magnitude (from the definition of a Gaussian distribution & wanting the objectness to be scaled correctly and the objectness temperature to be ~1). But you can defiantly think of this as being a hyperparameter that one tunes.

b) No, I only make 100 predictions per image. OW-DETR also makes 100 predictions per image:

    --output_dir ${EXP_DIR} --dataset owod **--num_queries 100** --eval_every 2 \
https://github.com/akshitac8/OW-DETR/blob/3515ff8c36687a4f582ef6a2c83866966b56ce23/configs/OWOD_split.sh#L9C50-L9C50

Best,
Orr

**Comment by Rzx520** [comment]
I see it, thanks.
re. loss_obj_ll, did you get it step by step through experiments? Could you please let me know some details about obtaining this parameter?@orrzohar

**Comment by Rzx520** [comment]
When I observed the data during the training process, I found that when it was set to 0.0008 and set to 0, the loss value only changed by about 1. I am a bit confused, but may this be a bit different for the category loss value? Haha, I really thought a bit too much. But miraculously, as the training progressed, I found that when the total loss value decreased by nearly double, the difference between 0.0008 and 0 was still around 1.@orrzohar

**Comment by orrzohar** [comment]
Re loss_obj_ll. The long story is:
I wanted the obj_temp to be scaled to be invariant of hidden_dim (dim. output of DDETR transformer decoder). So, in predicting a query embeddings objectness, I used:

```math
e^{-Z^2*\text{obj\_temp} / 256}
```

With Z^2 being the (normalized) distance from the mean (output from model), and obj_temp being the objectness temperature (see appendix of MS).



(again, this is all rough sketchpaper).


How does loss_obj_ll tie into this? the objectness loss is calculated as Z^2 * loss_obj_ll.
As you noted earlier, the overall loss is at the scale of 1 (for me, it conver. So, just assume that Z^2 * loss_obj_ll ~ 1 > Z^2~1/loss_obj_ll.

Lets put this in the equation above:

(lets call loss_obj_ll L):

```math
1/L * \text{obj\_temp} / 256 \sim \pi
```

```math
 L \sim \text{obj\_temp} / (256  * \pi )
```

I also wanted obj_temp~1. For this, I need: 

```math
 L \sim 1 / (256  * \pi ) \sim 0.001
```

From 0.001 to 0.0008 was pure experimentation because in the end, total_loss is composed of like 5 other losses which are balanced during optimization. But this is how I reached the general "scale" of this hyperparameter.

**Comment by orrzohar** [comment]
Re your second comment - loss_obj_ll is the objectness log-likelihood loss. If you turn it to 0, you will have poor objectness prediction (see ablation table in manuscript). 

I am not sure what your question is though, could you clarify?
